log4j-core/src/main/java/org/apache/logging/log4j/core/lookup/JndiLookup.java

```diff
@@ -40,6 +39,24 @@
     /** JNDI resource path prefix used in a J2EE container */
     static final String CONTAINER_JNDI_RESOURCE_PATH_PREFIX = "java:comp/env/";
 
+    public JndiLookup(){
+        // JNDI check
+        try {
+            // [LOG4J2-703] We might be on Android
+            Loader.newCheckedInstanceOf("org.apache.logging.log4j.core.lookup.JndiLookup", StrLookup.class);
```

Comment:
This is trying to load itself here?

Reply:
Hah, my fault. Your r right. How about just removing them? It is not necessary in terms of SPI mode.